Store.js

```diff
@@ -102,6 +112,11 @@ define(["dcl/dcl", "decor/Invalidating"], function (dcl, Invalidating) {
 		 */
 		computeProperties: function (props) {
 			if ("store" in props || "query" in props) {
+				if (!Array.isArray(this.store)) {
+					this._storeAdapter = new DstoreToStoreAdapter({store: this});
+				} else {
+					this._storeAdapter = new ArrayToStoreAdapter({store: this});
+				}
```

Comment:
I have done the modifications you asked in another branch. I haven't push it yet on the pull request that's why you can't see it here.